src/main/java/javaracingcar/JavaRacingCar.java

```diff
@@ -0,0 +1,94 @@
+package javaracingcar;
+
+import javaracingcar.entity.RacingCar;
+import javaracingcar.entity.RacingCarStepEnum;
+import javaracingcar.view.InputView;
+import javaracingcar.view.ResultView;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class JavaRacingCar {
+
+    private static final int RANDOM_VALUE_RANGE = 10;
+    private static final int DEFAULT_TRY_COUNT_VALUE = 0;
+
+    private final Random random = new Random();
+    private final List<RacingCar> cars = new ArrayList<>();
+
+    private int tryCount = DEFAULT_TRY_COUNT_VALUE;
+    private RacingCarStepEnum step = RacingCarStepEnum.INIT;
+
+    public JavaRacingCar() {
+
+    }
+
+    public void execution() {
+        requireCarCount(InputView.requireCarCount());
+        requireTryCount(InputView.requireTryCount());
+        start();
+    }
+
+    public void requireCarCount(int carCount) throws RuntimeException {
+        validateNegativeNumber(carCount);
+        initCars(carCount);
+        this.step = RacingCarStepEnum.CONFIRMED_CAR_COUNT;
+    }
+
+    public void requireTryCount(int tryCount) throws RuntimeException {
+        validateNegativeNumber(tryCount);
+        this.step = RacingCarStepEnum.CONFIRMED_TRY_COUNT;
+        this.tryCount = tryCount;
+    }
```

Comment:
```suggestion
    private void requireCarCount(int carCount) throws RuntimeException {
        validateNegativeNumber(carCount);
        initCars(carCount);
        this.step = RacingCarStepEnum.CONFIRMED_CAR_COUNT;
    }

    private void requireTryCount(int tryCount) throws RuntimeException {
        validateNegativeNumber(tryCount);
        this.step = RacingCarStepEnum.CONFIRMED_TRY_COUNT;
        this.tryCount = tryCount;
    }
```
public하게 열려있기보단 execution의 내부 기능이므로 private으로 막는 것이 좋을 것 같아요. 테스트를 위해서 열어두셨다면 private을 public하게 열기보다는 `execution()`을 호출하여 테스트하면 되지 않을까요?
해당 방법도 테스트하기 힘든 구조라면 객체를 분리하거나하는 방법을 생각해볼 수 있을 것 같네요 :)